cisagov/ScubaGear
# Make ScubaResults.json the default JSON output

## 🗣 Description ##



- Makes ScubaResults.json the default output for ScubaGear
- Inverts the `-MergeJSON` switch to `-KeepIndividualJSON` parameter. 
- Make `ScubaCached` backwards compatible with `ScubaResults.json`
- Update any tests that need it to match the new ScubaGear output behavior. 

TODO: 
- Update the functional and smoke tests to work with the ScubaResults.json in #1317 

## 💭 Motivation and context ##





Closes #1205
Closes #1023 

## 🧪 Testing ##





- Run `Invoke-SCuBA`. Observe `ScubaResults.json` is the default output.
- Run `Invoke-SCuBA -KeepIndividualJSON`. Observe `ProviderSettingsExport.json` is output instead of ScubaResutls.
- Run `Invoke-SCuBACached`. Observe `ScubaResults.json` is the default output.
- Run `Invoke-SCuBACached -ExportProvider $false`. Observe `ScubaCached` still performs as expected.
- Run `Invoke-SCuBACached` with the `-KeepIndividualJSON` flag. Observe performance benefit with this flag. Along with no change in behavior. 
- Run functional tests for a product as see no breaking changes.

- For reviewers and merge coordinator **Check smoke tests or other workflows to see if there are errors from the new output behavior**.


## ✅ Pre-approval checklist ##







## ✅ Pre-merge checklist ##






  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.

## ✅ Post-merge checklist ##
